src/dmd/expression.d

```diff
@@ -4827,11 +4827,19 @@ extern (C++) abstract class BinExp : Expression
         TOK thisOp = (op == TOKquestion) ? TOKcolon : op;
         if (e1.op == TOKtype || e2.op == TOKtype)
         {
-            error("incompatible types for ((%s) %s (%s)): cannot use '%s' with types", e1.toChars(), Token.toChars(thisOp), e2.toChars(), Token.toChars(op));
+            error("incompatible types for ((%s) %s (%s)): cannot use '%s' with types",
+                e1.toChars(), Token.toChars(thisOp), e2.toChars(), Token.toChars(op));
+        }
+        else if (e1.type.equals(e2.type))
+        {
+            error("incompatible types for ((%s) %s (%s)): both operands are of type '%s'",
```

Comment:
I think it would be better to denote that _the operator %s cant be used for two variable of type %s_.
But maybe in another PR ?

Reply:
Reworded in f73a72b.